src/main_thread/tracks_store/tracks_store.ts

```diff
@@ -481,17 +498,119 @@ export default class TracksStore extends EventEmitter<ITracksStoreEvents> {
       if (
         periodObj.isPeriodAdvertised &&
         trackObj.dispatcher !== null &&
-        !trackObj.dispatcher.hasSetTrack() &&
-        trackObj.storedSettings !== undefined
+        !trackObj.dispatcher.hasSetTrack()
       ) {
-        trackObj.dispatcher.updateTrack(trackObj.storedSettings);
+        trackObj.dispatcher.updateTrack(trackObj.storedSettings ?? null);
```

Comment:
I'm not sure I get it here.

To me, if you don't want a type of media, `storedSettings` should always be set explicitely to `null`. If `storedSettings` is ever `undefined`, it should always mean that we don't know yet